### PR TITLE
allow DataOutputAgent to build complex XML

### LINKED ISSUE #834: In DataOutputAgent, enclosed media cannot be output
I have a source RSS feed that I want to filter before importing it in my favorite Podcast Player on Android. The issue is that I can't add attributes to entries in the xml output of an item. This is the output I want to be able to achieve :

```
<rss version="2.0">
<channel>
<ttl>360</ttl>
<link>http://example.com</link>
<title>A title</title>
<description>A description</description>
<item>
<title>Item title</title>
<link>http://example.com</link>
<enclosure url="http://something.com/1.mp3" length="9502596" type="audio/mpeg"/>
</item>
```

Look at the "enclosure" line : it has 3 mandatory attributes to link to the content.
How to achieve that ?
Thanks

### PR DESCRIPTION
**Pull request by cantino**
Should fix #834.  Appreciate review!

### CODE REVIEW

**Comment by cantino** [comment]
@dsander Mind reviewing this?

File: spec/models/agents/wunderlist_agent_spec.rb [hunk 1]
```diff
@@ -54,8 +54,9 @@
 
   describe "#receive" do
     it "send a message to the hipchat" do
-      stub_request(:post, 'https://a.wunderlist.com/api/v1/tasks').with { |request| request.body == 'abc'}
+      stub_request(:post, 'https://a.wunderlist.com/api/v1/tasks')
       @checker.receive([@event])
+      expect(WebMock).to have_requested(:post, "https://a.wunderlist.com/api/v1/tasks")
```
Comment (dsander) [inline spec/models/agents/wunderlist_agent_spec.rb]: Are those changes related?

**Comment by dsander** [comment]
The code looks good to me.

File: spec/models/agents/wunderlist_agent_spec.rb (hunk 1 above)
Comment (cantino) [inline spec/models/agents/wunderlist_agent_spec.rb]: I didn't see any indication that the `.with { |request| request.body == 'abc'}` part was doing anything, or that that boolean expression had any effect.

**Pull request merged**